crates/model/src/order.rs

```diff
@@ -40,6 +41,7 @@ pub struct Order {
     pub data: OrderData,
     #[serde(flatten)]
     pub signature: Signature,
+    pub pre_interactions: Vec<InteractionData>,
```

Comment:
What component will need to access them? I think maybe the normal api shouldn't be dealing with these which would make it synergise with https://github.com/cowprotocol/services/pull/529 which allows us to have a separate struct for orders used by autopilot, driver vs api.

Reply:
Autopilot: For checks like sufficient approval, the interactions are needed
Driver: For building the settlement
Solver: Not sure... maybe they wanna check something as well. Maybe the gas costs?
Orderbook: I think at some point we wanna serve the pre-interactions via the API and show them in the UI. But its really not clear or very vague.

Reply:
> Solver: Not sure... maybe they wanna check something as well. Maybe the gas costs?

They need it for building the settlement calldata right?